improver_tests/precipitation_type/hail_fraction/test_HailFraction.py

```diff
@@ -0,0 +1,197 @@
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Tests for the HailFraction plugin."""
+
+import iris
+import numpy as np
+import pytest
+from cf_units import Unit
+
+from improver.precipitation_type.hail_fraction import HailFraction
+from improver.synthetic_data.set_up_test_cubes import set_up_variable_cube
+
+COMMON_ATTRS = {
+    "source": "Unit test",
+    "institution": "Met Office",
+    "title": "Post-Processed IMPROVER unit test",
+}
+
+
+def setup_cubes():
+    """Set up cubes for testing."""
+    vertical_updraught_data = np.zeros((2, 2), dtype=np.float32)
+    vertical_updraught = set_up_variable_cube(
+        vertical_updraught_data,
+        "maximum_vertical_updraught",
+        "m s-1",
+        spatial_grid="equalarea",
+        attributes=COMMON_ATTRS,
+        standard_grid_metadata="gl_ens",
+    )
+
+    hail_size_data = np.zeros((2, 2), dtype=np.float32)
+    hail_size = set_up_variable_cube(
+        hail_size_data,
+        "size_of_hail_stones",
+        "m",
+        spatial_grid="equalarea",
+        attributes=COMMON_ATTRS,
+        standard_grid_metadata="gl_ens",
+    )
+    hail_size.convert_units("m")
+
+    cloud_condensation_level_data = np.zeros((2, 2), dtype=np.float32)
+    cloud_condensation_level = set_up_variable_cube(
+        cloud_condensation_level_data,
+        "air_temperature_at_condensation_level",
+        "K",
+        spatial_grid="equalarea",
+        attributes=COMMON_ATTRS,
+        standard_grid_metadata="gl_ens",
+    )
+    cloud_condensation_level.convert_units("Kelvin")
```

Comment:
This statement relabels the units from "K" to "Kelvin". Why not just set this on line 75?